Objects/obmalloc.c

```diff
@@ -2666,6 +2668,40 @@ _PyDebugAllocatorStats(FILE *out,
 
 #ifdef WITH_PYMALLOC
 
+static void
+free_arenas(PyInterpreterState *interp)
+{
+    OMState *state = &interp->obmalloc;
+    for (uint i = 0; i < maxarenas; ++i) {
+        // free each obmalloc memory arena
+        struct arena_object *ao = &allarenas[i];
+        _PyObject_Arena.free(_PyObject_Arena.ctx,
+                             (void *)ao->address, ARENA_SIZE);
+    }
+    // free the array containing pointers to all arenas
+    PyMem_RawFree(allarenas);
+#if WITH_PYMALLOC_RADIX_TREE
+#ifdef USE_INTERIOR_NODES
+    // Free the middle and bottom nodes of the radix tree.  These are allocated
+    // by arena_map_mark_used() but not freed when arenas are freed.
```

Comment:
Better performance and simpler code.  It might not be too expensive to free them.  We would have to track how many nodes are children and then remove when count reaches zero.